### PR TITLE
Snap96 Keyboard

### PR DESCRIPTION
**Pull request by Xelus22**
## Description
A new keyboard that is based on 96 which can "snap" to smaller layouts. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* New keyboard

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/xelus/snap96/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,31 @@
+// Default layout for Snap96
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT_all(
+        KC_ESC,  KC_F1,   KC_F2,   KC_F3,  KC_F4,  KC_F5, KC_F6,  KC_F7,  KC_F8,  KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_INS,  KC_DEL,  KC_PSCR, KC_SLCK, KC_HOME, KC_END,
+		KC_GRV,  KC_1,    KC_2,    KC_3,   KC_4,   KC_5,  KC_6,   KC_7,   KC_8,   KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_BSPC, KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
+		KC_TAB,  KC_Q,    KC_W,    KC_E,   KC_R,   KC_T,  KC_Y,   KC_U,   KC_I,   KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,          KC_P7,   KC_P8,   KC_P9,   KC_PPLS,
+		KC_CAPS, KC_A,    KC_S,    KC_D,   KC_F,   KC_G,  KC_H,   KC_J,   KC_K,   KC_L,    KC_SCLN, KC_QUOT, KC_ENT,                    KC_P4,   KC_P5,   KC_P6,   KC_PPLS,
+		KC_LSFT, KC_PIPE, KC_Z,    KC_X,   KC_C,   KC_V,  KC_B,   KC_N,   KC_M,   KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,          KC_UP,   KC_P1,   KC_P2,   KC_P3,   KC_PENT,
```
Comment (noroadsleft) [inline keyboards/xelus/snap96/keymaps/default/keymap.c]: This should probably be Non-US Backslash instead.

```suggestion
		KC_LSFT, KC_NUBS, KC_Z,    KC_X,   KC_C,   KC_V,  KC_B,   KC_N,   KC_M,   KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,          KC_UP,   KC_P1,   KC_P2,   KC_P3,   KC_PENT,
```

File: keyboards/xelus/snap96/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,31 @@
+// VIA layout for Snap96
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT_all(
+        KC_ESC,  KC_F1,   KC_F2,   KC_F3,  KC_F4,  KC_F5, KC_F6,  KC_F7,  KC_F8,  KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_INS,  KC_DEL,  KC_PSCR, KC_SLCK, KC_HOME, KC_END,
+		KC_GRV,  KC_1,    KC_2,    KC_3,   KC_4,   KC_5,  KC_6,   KC_7,   KC_8,   KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_BSPC, KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
+		KC_TAB,  KC_Q,    KC_W,    KC_E,   KC_R,   KC_T,  KC_Y,   KC_U,   KC_I,   KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,          KC_P7,   KC_P8,   KC_P9,   KC_PPLS,
+		KC_CAPS, KC_A,    KC_S,    KC_D,   KC_F,   KC_G,  KC_H,   KC_J,   KC_K,   KC_L,    KC_SCLN, KC_QUOT, KC_ENT,                    KC_P4,   KC_P5,   KC_P6,   KC_PPLS,
+		KC_LSFT, KC_PIPE, KC_Z,    KC_X,   KC_C,   KC_V,  KC_B,   KC_N,   KC_M,   KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,          KC_UP,   KC_P1,   KC_P2,   KC_P3,   KC_PENT,
```
Comment (noroadsleft) [inline keyboards/xelus/snap96/keymaps/via/keymap.c]:
```suggestion
		KC_LSFT, KC_NUBS, KC_Z,    KC_X,   KC_C,   KC_V,  KC_B,   KC_N,   KC_M,   KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,          KC_UP,   KC_P1,   KC_P2,   KC_P3,   KC_PENT,
```

File: keyboards/xelus/snap96/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+    [2] = LAYOUT_all(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS,                            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    )
+};
```
Comment (drashna) [inline keyboards/xelus/snap96/keymaps/default/keymap.c]: These look like they're all transparent?   If so, they can be removed. 

Also, you may want to consider using `_______` (that's 7 underscores) for transparent, as it's alias and makes it easier to read.

Comment (Xelus22) [reply keyboards/xelus/snap96/keymaps/default/keymap.c]: I'll remove layer 2 but i think keeping just one layer empty will be more beneficial for those in future who wish to add and edit their own keymap in.

File: keyboards/xelus/snap96/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    ),
+
+    [1] = LAYOUT_all(
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,                   _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______, _______,
+        _______, _______, _______,                            _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______, _______, _______
+    )
```
Comment (zvecr) [inline keyboards/xelus/snap96/keymaps/via/keymap.c]: Given you set DYNAMIC_KEYMAP_LAYER_COUNT to 3,
```suggestion
    ),

    [2] = LAYOUT_all(
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,                   _______, _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______, _______,
        _______, _______, _______,                            _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______, _______, _______
    )
```

File: keyboards/xelus/snap96/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,24 @@
+// Default layout for Snap96
+#include QMK_KEYBOARD_H
+
+#define _______ KC_TRNS
+
```
Comment (noroadsleft) [inline keyboards/xelus/snap96/keymaps/default/keymap.c]: QMK defines this automatically.

https://github.com/qmk/qmk_firmware/blob/e34764502f7fdd0e1104b18fb4ddf77a352ee2a7/quantum/quantum_keycodes.h#L29-L31

```suggestion
```

File: keyboards/xelus/snap96/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,24 @@
+// VIA layout for Snap96
+#include QMK_KEYBOARD_H
+
+#define _______ KC_TRNS
+
```
Comment (noroadsleft) [inline keyboards/xelus/snap96/keymaps/via/keymap.c]:
```suggestion
```

**Pull request merged**